strawberryfields/gbs/sample.py

```diff
@@ -332,37 +372,70 @@ def gaussian(
     if not 0 <= loss <= 1:
         raise ValueError("Loss parameter must take a value between zero and one")
 
-    n_modes = len(alpha)
+    n_modes = len(t)
+
+    # initialize engine
 
     eng = sf.LocalEngine(backend="gaussian")
-    gbs = sf.Program(n_modes)
 
+    # initialize program objects
+
+    if np.all(t != 0):
+        gbs = sf.Program(n_modes * 2)
+    else:
+        gbs = sf.Program(n_modes)
+
+    # start circuit
     # pylint: disable=expression-not-assigned,pointless-statement
     with gbs.context as q:
 
-        # this interferometer has no action, but is kept to follow the decomposition given in the
-        # docstring
-        sf.ops.Interferometer(U1) | q
+        # two-mode squeezing gates
+
+        if np.all(t != 0):
+            for i in range(n_modes):
+                sf.ops.S2gate(t[i]) | (q[i], q[i + n_modes])
+
+        # first interferometer
+
+        sf.ops.Interferometer(U1) | q[:n_modes]
+
+        # squeezing gates
 
         for i in range(n_modes):
             sf.ops.Sgate(r[i]) | q[i]
 
-        sf.ops.Interferometer(U2) | q
+        # second interferometer
+
+        sf.ops.Interferometer(U2) | q[:n_modes]
+
+        # displacement gates
 
         for i in range(n_modes):
             sf.ops.Dgate(alpha[i]) | q[i]
 
+        # apply loss
+
         if loss:
             for _q in q:
                 sf.ops.LossChannel(1 - loss) | _q
 
+        # measurement
+
         sf.ops.MeasureFock() | q
 
     with warnings.catch_warnings():
         warnings.filterwarnings("ignore", category=UserWarning, message="Cannot simulate non-")
+
+        # run the engine and generate samples
+
         s = eng.run(gbs, run_options={"shots": n_samples}).samples
 
-    if n_samples == 1:
-        return [s]
+    # end circuit
 
-    return s.tolist()
+    s = np.array(s).tolist() # convert all generated samples to list
```

Comment:
We should have it because the sampler is inconsistent in the format of the returned samples:
For one single sample it give you a list while for more than one sample it gives you an array. If we remove this line, then we have to add an extra if condition to deal with that, otherwise we get list to list conversion error.